File: isplitapp-core/core/Expenses/Payloads/PartyInfo.cs
using System.Text.Json.Serialization;
using IB.ISplitApp.Core.Utils;

namespace IB.ISplitApp.Core.Expenses.Payloads;

/// <summary>
/// Response with party data and additional summary 
/// </summary>
public record PartyInfo
{
    /// <summary>
    /// Unique party ID 
    /// </summary>
    public string Id { get; init; } = IdUtil.DefaultId;
    
    /// <summary>
    /// Name of the party
    /// </summary>
    public string Name { get; init; } = string.Empty;
    
    /// <summary>
    /// Name of the currency for all expenses in the group
    /// </summary>
    public string Currency { get; init; } = "USD";
    
    /// <summary>
    /// When the party was created in UTC time
    /// </summary>
    public DateTime Created { get; init; } = DateTime.UtcNow;
    
    /// <summary>
    /// Last time the party was updated in UTC
    /// </summary>
    public DateTime Updated { get; init; } = DateTime.UtcNow;
    
    /// <summary>
    /// How much money the party has spent in total
    /// </summary>
    [JsonPropertyName("totalExpenses")]
    public decimal FuTotalExpenses { get; init; } = 0;
    
    /// <summary>
    /// Total amount of the party members
    /// </summary>
    public int TotalParticipants { get; init; } = 0;

    /// <summary>
    /// Number of expenses
    /// </summary>
    public int TotalExpenseNumber { get; init; } = 0;

    /// <summary>
    /// Total unpaid balance
    /// </summary>
    [JsonPropertyName("totalBorrow")]
    public decimal FuTotalBorrow { get; init; } = 0;
    
    /// <summary>
    /// List of all participants in the party
    /// </summary>
    public ParticipantResponse[] Participants { get; init; } = [];
}

File: next-ui/src/api/contract/PartyInfo.ts
/* eslint-disable @typescript-eslint/no-inferrable-types */
import { ParticipantInfo } from "./ParticipantInfo";

/**
 *  Full information about the party
 */
export class PartyInfo {
    id: string = "";
    
    name: string = "";

    currency: string = "";
    
    created: Date = new Date();

    updated: Date = new Date();
    
    totalExpenses: number = 0;

    totalTransactions: number = 0;

    outstandingBalance: number = 0;

    totalParticipants: number = 0;
    
    participants: ParticipantInfo[] = [];

    isArchived: boolean = false;

    updateTimestamp = "zzzzzzzzz";
}

